protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java

```diff
@@ -334,28 +334,28 @@ private void handleResponseException(ImapProcessor.Responder responder, MailboxE
         responder.respond(response);
     }
 
-    protected void okComplete(ImapCommand command, Tag tag, ImapProcessor.Responder responder) {
-        final StatusResponse response = factory.taggedOk(tag, command, HumanReadableText.COMPLETED);
+    protected void okComplete(ImapRequest request, ImapProcessor.Responder responder) {
+        final StatusResponse response = factory.taggedOk(request.getTag(), request.getCommand(), HumanReadableText.COMPLETED);
         responder.respond(response);
     }
 
-    protected void okComplete(ImapCommand command, Tag tag, ResponseCode code, ImapProcessor.Responder responder) {
-        final StatusResponse response = factory.taggedOk(tag, command, HumanReadableText.COMPLETED, code);
+    protected void okComplete(ImapRequest request,  ResponseCode code, ImapProcessor.Responder responder) {
+        final StatusResponse response = factory.taggedOk(request.getTag(), request.getCommand(), HumanReadableText.COMPLETED, code);
```

Comment:
oh I see... you are getting it from the request.
It is always right? I believe is some cases (unsolicited response) the tag could be different than the request's one, no?